demo/test-extensions/Makefile

```diff
@@ -0,0 +1,18 @@
+CFLAGS ?= -Os -Wall
+
+PKG_MODULES   := gobject-2.0 webkit2gtk-web-extension-4.0
+WEB_EXT_FLAGS := $(shell pkg-config ${PKG_MODULES} --cflags)
+WEB_EXT_LIBS  := $(shell pkg-config ${PKG_MODULES} --libs)
+
+CPPFLAGS += ${WEB_EXT_FLAGS}
+LDLIBS   += ${WEB_EXT_LIBS}
+
+all: test-extension.so
+
+test-extension.so: test-extension.o
+	${LD} ${LDFLAGS} -fPIC -shared -o $@ $^ ${LDLIBS}
+
+test-extension.o: CFLAGS += -fPIC
```

Comment:
Do you need "-fPIC" here?  I'm not sure but if I recall correctly packages may want to decide if they want to pass "-fPIC" or not.

Reply:
Is `target: variable-settings` valid syntax?

Reply:
I remember reading that `-fPIC` is needed here, but I cannot recall, what the reason was. 

It works without this flag, though.

The syntax is valid at least in GNU `make` -- there are so-called [Implicit Rules](https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_chapter/make_10.html) that `make` uses in case no command was provided.